packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -129,6 +129,7 @@ async function migrateToArmAndMultiEnv(ctx: CoreHookContext): Promise<void> {
     await cleanup(projectPath);
     throw err;
   }
+  await backup(projectPath);
```

Comment:
Unless migrtation success, will not remove or change user files. that's a principle. 

As you say that "If other logic is inserted in the middle before backup, which break the files or adjust the files' contents.", if other logic break or adjust the files before backup and remove, that mean user config is broken when migration fails(permission, network, shutdown..). 
So, before migration success, we shouldn't remove or change user config.

Reply:
Currently our logic is guaranteed by you and me, but there is no guarantee that others will change this part of the logic in the future. In case someone accidentally changes the content of the file before backing up, the files need to be backed up is no longer the source file.